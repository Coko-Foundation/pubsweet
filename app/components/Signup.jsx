// import { Field } from 'react-redux-form'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { signupUser } from '../actions'
import React, { Component, PropTypes } from 'react'

class Signup extends Component {
  render () {
    return (
      <div>
        <input type='text' ref='username' className='form-control' placeholder='Username'/>
        <input type='text' ref='email' className='form-control' placeholder='Email'/>
        <input type='password' ref='password' className='form-control' placeholder='Password'/>
        <button onClick={(event) => this.handleClick(event)} className='btn btn-primary'>
          Sign up
        </button>
      </div>
    )
  }

  handleClick (event) {
    const user = {
      username: this.refs.username.value,
      email: this.refs.email.value,
      password: this.refs.password.value
    }
    this.props.actions.signupUser(user)
  }
}

Signup.propTypes = {
    // Dispatch
  actions: PropTypes.object
}

function mapState (state) {
  return { auth: state.auth }
}

function mapDispatch (dispatch) {
  return {
    actions: bindActionCreators({ signupUser }, dispatch)
  }
}

export default connect(
  mapState, mapDispatch
)(Signup)
