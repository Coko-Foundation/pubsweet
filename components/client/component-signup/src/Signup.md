A signup form

```js
const { withFormik } = require('formik')

const SignupForm = withFormik({
  initialValues: {
    username: '',
    password: '',
  },
  mapPropsToValues: props => ({
    username: props.username,
    password: props.password,
  }),
  displayName: 'Signup',
  handleSubmit: val => console.log(val),
})(Signup)
;<SignupForm />
```

Can also have a logo:

```js
const { withFormik } = require('formik')

const SignupForm = withFormik({
  initialValues: {
    username: '',
    password: '',
  },
  mapPropsToValues: props => ({
    username: props.username,
    password: props.password,
  }),
  displayName: 'Signup',
  handleSubmit: val => console.log(val),
})(Signup)
;<SignupForm logo="pubsweet-logo.svg" />
```
