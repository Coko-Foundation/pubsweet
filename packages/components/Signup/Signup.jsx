import React from 'react'
import { Field } from 'formik'
import { isEmpty } from 'lodash'
import {
  CenteredColumn,
  Link,
  H1,
  ErrorText,
  Button,
  TextField,
} from '@pubsweet/ui'

const UsernameInput = props => <TextField label="Username" {...props.field} />
const EmailInput = props => (
  <TextField label="Email" {...props.field} type="email" />
)
const PasswordInput = props => (
  <TextField label="Password" {...props.field} type="password" />
)

const Signup = ({ errors, handleSubmit }) => (
  <CenteredColumn small>
    <H1>Sign up</H1>

    {!isEmpty(errors) && <ErrorText>{errors}</ErrorText>}

    <form onSubmit={handleSubmit}>
      <Field component={UsernameInput} name="username" />
      <Field component={EmailInput} name="email" />
      <Field component={PasswordInput} name="password" />
      <Button primary type="submit">
        Sign up
      </Button>
    </form>

    <div>
      <span>Already have an account? </span>
      <Link to="/login">Login</Link>
    </div>
  </CenteredColumn>
)

export default Signup
