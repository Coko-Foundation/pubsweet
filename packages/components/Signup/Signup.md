A Submit form

```js
const { withFormik } = require('formik')

const SignupForm = withFormik({
  initialValues: {
    username: '',
    password: '',
    email: '',
  },
  mapPropsToValues: props => ({
    username: props.username,
    email: props.email,
    password: props.password,
  }),
  displayName: 'signup',
  handleSubmit,
})(Signup)
;<SignupForm />
```

Which can have an error message:

```js
const { withFormik } = require('formik')

const SignupForm = withFormik({
  initialValues: {
    username: '',
    password: '',
    email: '',
  },
  mapPropsToValues: props => ({
    username: props.username,
    email: props.email,
    password: props.password,
  }),
  displayName: 'signup-error',
  handleSubmit: (values, { setErrors }) => setErrors('Error message'),
})(Signup)
;<SignupForm />
```
