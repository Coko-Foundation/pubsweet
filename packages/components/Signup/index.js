module.exports = {
  client: {
    components: [() => require('./src/SignupContainer')],
  },
}
