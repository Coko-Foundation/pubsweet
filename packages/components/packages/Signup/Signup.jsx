import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Grid, Alert, Row, Col } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import styles from './Signup.local.scss'

export default class Signup extends Component {
  constructor(props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick(event) {
    event.preventDefault()
    const user = {
      username: this.usernameElement.value,
      email: this.emailElement.value,
      password: this.passwordElement.value,
    }
    this.props.actions.signupUser(user)
  }

  render() {
    const self = this
    const { error } = self.props
    self.refs = {}
    return (
      <div className="bootstrap" style={{ marginTop: 20 }}>
        <Grid>
          <Row>
            <Col md={2} mdOffset={5}>
              <img
                alt="pubsweet-logo"
                className={styles.logo}
                src="/assets/pubsweet-rgb-small.jpg"
                style={{ maxWidth: '100%' }}
              />
            </Col>
          </Row>

          <Row>
            <Col md={4}>
              {error && (
                <Alert bsStyle="warning">
                  <i className="fa fa-exclamation-circle" />&nbsp; {error}
                </Alert>
              )}
            </Col>

            <Col className={styles.signup} md={4} xs={12}>
              <h1>Sign up</h1>
              <form>
                <div className="form-group">
                  <label htmlFor="username">Username</label>
                  <input
                    className="form-control"
                    placeholder="Username"
                    ref={el => (this.usernameElement = el)}
                    type="text"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="email">Email</label>
                  <input
                    className="form-control"
                    placeholder="Email"
                    ref={el => (this.emailElement = el)}
                    type="text"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <input
                    className="form-control"
                    placeholder="Password"
                    ref={el => (this.passwordElement = el)}
                    type="password"
                  />
                </div>
                <button
                  className={`${styles.button} btn btn-block btn-primary`}
                  onClick={this.handleClick}
                >
                  Sign up
                </button>
                <p>
                  Already have an account?<br />
                  <Link to="/login">Log in here</Link>
                </p>
              </form>
            </Col>
          </Row>
        </Grid>
      </div>
    )
  }
}

Signup.propTypes = {
  actions: PropTypes.object,
}
