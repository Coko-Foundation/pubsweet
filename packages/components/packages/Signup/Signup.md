A login form

```js
const { reduxForm } = require('redux-form')

const SignupForm = reduxForm({
  form: 'signup',
  onChange: values => console.log(values),
})(Signup)
;<SignupForm />
```

Which can have an error message:

```js
const { reduxForm } = require('redux-form')

const SignupForm = reduxForm({
  form: 'signup-error',
  onSubmit: val => console.log(val),
})(Signup)
;<SignupForm error={'Yikes!'} />
```
