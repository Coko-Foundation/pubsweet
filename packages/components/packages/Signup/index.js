module.exports = {
  frontend: {
    components: [() => require('./SignupContainer')],
    actions: () => require('./actions'),
    reducers: () => require('./reducers'),
  },
}
