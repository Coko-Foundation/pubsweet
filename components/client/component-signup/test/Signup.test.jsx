import { shallow } from 'enzyme'
import React from 'react'
import { Field } from 'formik'
import { Button, ErrorText } from '@pubsweet/ui'

import Signup from '../src/Signup'

describe('<Signup />', () => {
  const makeWrapper = (props = {}) => shallow(<Signup {...props} />)

  it('renders the login form', () => {
    const wrapper = makeWrapper()
    expect(wrapper.debug()).toMatchSnapshot()
    expect(wrapper.find(Field)).toHaveLength(3)
    expect(wrapper.find(Button)).toHaveLength(1)
    expect(wrapper.find({ to: '/login' })).toHaveLength(1)
  })

  it('shows error', () => {
    const wrapper = makeWrapper({ status: { error: 'Yikes!' } })
    expect(wrapper.find(ErrorText)).toHaveLength(1)
  })

  it('can hide logo', () => {
    const logo = 'data:image/gif;base64,R0lGODlhDwAPAKECAAAAzMzM/////'
    const wrapper1 = makeWrapper({ logo })
    const wrapper2 = makeWrapper()
    expect(wrapper1.find('Logo')).toHaveLength(1)
    expect(wrapper2.find('Logo')).toHaveLength(0)
  })

  it('triggers submit handler', () => {
    const handleSubmit = jest.fn()
    const wrapper = makeWrapper({ handleSubmit })
    wrapper.find('form').simulate('submit')
    expect(handleSubmit).toHaveBeenCalled()
  })
})
