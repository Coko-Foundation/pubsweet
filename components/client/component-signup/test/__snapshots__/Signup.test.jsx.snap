// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders the login form 1`] = `
"<styled.div small={true}>
  <styled.div>
    <styled.h1>
      Sign up
    </styled.h1>
    <form onSubmit={[undefined]}>
      <FormikConnect(FieldInner) component={[Function: UsernameInput]} name=\\"username\\" />
      <FormikConnect(FieldInner) component={[Function: EmailInput]} name=\\"email\\" />
      <FormikConnect(FieldInner) component={[Function: PasswordInput]} name=\\"password\\" />
      <styled.button primary={true} type=\\"submit\\">
        Sign up
      </styled.button>
    </form>
    <div>
      <span>
        Already have an account? 
      </span>
      <Styled(SmartUnstyledLink) to=\\"/login\\">
        Login
      </Styled(SmartUnstyledLink)>
    </div>
  </styled.div>
</styled.div>"
`;
